src/Orleans/Core/GrainAttributes.cs

```diff
@@ -86,6 +87,37 @@ public sealed class ImmutableAttribute : Attribute
         }
     }
 
+    namespace MultiCluster
+    {
+        /// <summary>
+        /// base class for multi cluster registration strategies.
+        /// </summary>
+        public abstract class RegistrationAttribute : Attribute
```

Comment:
Should it have AttributeUsage to say it's only at the class (or interface?) level and that multiple times is not allowed?